samples/drivers/crypto/src/main.c

```diff
@@ -213,7 +213,7 @@ void ctr_mode(void)
 	struct cipher_pkt decrypt;
 	u8_t encrypted[64] = {0};
 	u8_t decrypted[64] = {0};
-	u8_t iv[12] = {
+	u8_t iv1[12] = {
```

Comment:
I prefer global variables to have larger, more descriptive names, and local variables (including parameters) to have shorter names.  The rationale is that global variables are often declared at the top of the file, requiring one to scroll to see it; it's rarely in the programmer's mental context when writing/reading code.

I'm not strongly opposed to adding suffixes to local variables in order to fix the shadow warnings (it's kind of a  tomayto tomahto situation), but consider renaming the global `iv` variable instead.

Reply:
oh yeah, I remember now, thinking about what I should call the variables was the hardest part of this patch :)

Reply:
The global `iv` could be moved to the function `cbc_mode()` since it's only used there. The `ctr_mode()` function uses its own initialization vector.